packages/core/helper-plugin/lib/src/components/GenericInput/index.js

```diff
@@ -44,12 +44,30 @@ const GenericInput = ({
 }) => {
   const { formatMessage } = useIntl();
   const [showPassword, setShowPassword] = useState(false);
-
-  const CustomInput = customInputs ? customInputs[type] : null;
-
   // the API always returns null, which throws an error in React,
   // therefore we cast this case to undefined
   const value = defaultValue ?? undefined;
+  const CustomInput = customInputs ? customInputs[type] : null;
+
+  function getErrorMessage(error) {
+    if (!error) {
+      return null;
+    }
+
+    if (typeof error === 'string') {
+      return formatMessage({ id: error, defaultMessage: error });
+    }
+
+    return formatMessage(
+      {
+        id: error.id,
+        defaultMessage: error?.defaultMessage ?? error.id,
+      },
+      { fieldName: formatMessage(intlLabel) }
```

Comment:
I don't think the GenericInput component should handle errors format to both string and object. You are trying to make an update that match your specific use case and I think it increases the complexity of this component.

What you could do is to define the message differently:

```js
formatMessage({ id: error.id, defaultMessage: error.defaultMessage }, errors.values)
```

Reply:
I just invented that, so that I don't have to refactor the error handling in the whole admin app. I could go down that path, but it will take much longer and needs a lot of testing ...

Reply:
Ah I see. 🤔 The problem though is, that we don't have the info about the intl field name, at the time the error is thrown and I'd like to rather not move up this logic. I could also remove the `fieldName` entirely - it was added because most other error messages mention the field name in the message instead of static one ("name contains error" vs. "field contains error").

Reply:
You can still introduce the `error.values` it's a pattern we already use in the codebase like [here](https://github.com/strapi/strapi/blob/7b81724c5abe7b5360f7be569a3a51a6df8a557f/packages/core/admin/admin/src/components/Notifications/Notification/index.js#L11). This way you can still pass dynamic variables.

Reply:
I've done that now on d3ca3746b134e48e34bceeabb1b7207448f67c05 . Overall I like it and it's less opinionated, but we are not showing the label anymore (because that is not available at the time the error is thrown), but the field name instead. I can live with that.

What do you think?